rero_ils/modules/loans/api.py

```diff
@@ -159,23 +159,23 @@ def organisation_pid(self):
         if self.get('item_pid'):
             item = Item.get_record_by_pid(self.get('item_pid'))
             return item.organisation_pid
-        return None
 
     @property
     def library_pid(self):
-        """Get library PID regarding transaction location PID or location."""
+        """Get library PID regarding loan location."""
+        return Location.get_record_by_pid(self.location_pid).library_pid
+
+    @property
+    def location_pid(self):
+        """Get loan transaction_location PID or item owning location."""
         from ..items.api import Item
 
         location_pid = self.get('transaction_location_pid')
+        if location_pid:
+            return location_pid
```

Comment:
you can rewrite something like:
```python

return self.get('transaction_location_pid', 
                            Item.get_record_by_pid(self['item_pid']).holding_location_pid
                       )
```

Reply:
Don't know very well the circulation module, but this return statement should return an exception if `transaction_location_pid` and `item_pid` are undefined.

Reply:
As suggested, I used `try...except` pattern